daemon/containerd/image_list.go

```diff
@@ -5,21 +5,36 @@ import (
 
 	"github.com/containerd/containerd"
 	"github.com/containerd/containerd/snapshots"
+	"github.com/docker/distribution/reference"
 	"github.com/docker/docker/api/types"
 	"github.com/opencontainers/image-spec/identity"
 )
 
+type imageFilterFunc func(image containerd.Image) bool
+
 // Images returns a filtered list of images.
 func (i *ImageService) Images(ctx context.Context, opts types.ImageListOptions) ([]*types.ImageSummary, error) {
 	imgs, err := i.client.ListImages(ctx)
```

Comment:
You can probably do the filters as fieldpaths in this call.

Reply:
https://github.com/containerd/containerd/blob/main/filters/filter.go

Doing the date comparison is not possible today, but most things should be.

Reply:
Oh, that looks interesting indeed. Will have to dig into that a bit to see how to integrate it (I'm mostly upstreaming patches here 😂)

what's your preference? trying to make that work before we merge this, or improve in a follow-up?

Reply:
I'm fine with it as is, let's maybe make sure to note though that the implementation needs work.

Reply:
I added some todo's and tracking issues;

- https://github.com/moby/moby/issues/43845
- https://github.com/moby/moby/issues/43846
- https://github.com/moby/moby/issues/43847
- https://github.com/moby/moby/issues/43848